net-libs/libbitcoinconsensus/libbitcoinconsensus-0.21.1.ebuild

```diff
@@ -0,0 +1,111 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools
+
+BITCOINCORE_COMMITHASH="194b9b8792d9b0798fdb570b79fa51f1d1f5ebaf"
+KNOTS_PV="${PV}.knots20210629"
+KNOTS_P="bitcoin-${KNOTS_PV}"
+
+DESCRIPTION="Bitcoin Core consensus library"
+HOMEPAGE="https://bitcoincore.org/ https://bitcoinknots.org/"
+SRC_URI="
+	https://github.com/bitcoin/bitcoin/archive/${BITCOINCORE_COMMITHASH}.tar.gz -> bitcoin-v${PV}.tar.gz
+	https://bitcoinknots.org/files/0.21.x/${KNOTS_PV}/${KNOTS_P}.patches.txz -> ${KNOTS_P}.patches.tar.xz
+"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="+asm bitcoin_protocol_taproot knots"
+
+DEPEND="
+	>dev-libs/libsecp256k1-0.1_pre20200911:=[recovery,schnorr]
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	>=sys-devel/autoconf-2.69
+	>=sys-devel/automake-1.13
+"
+
+DOCS=( doc/bips.md doc/release-notes.md doc/shared-libraries.md )
+
+S="${WORKDIR}/bitcoin-${BITCOINCORE_COMMITHASH}"
+
+pkg_pretend() {
+	if use knots; then
+		elog "You are building ${PN} from Bitcoin Knots."
+		elog "For more information, see:"
+		elog "https://bitcoinknots.org/files/0.21.x/${KNOTS_PV}/${KNOTS_P}.desc.html"
+	else
+		elog "You are building ${PN} from Bitcoin Core."
+		elog "For more information, see:"
+		elog "https://bitcoincore.org/en/2021/05/01/release-${PV}/"
+	fi
+	elog
+	ewarn "CAUTION: BITCOIN PROTOCOL CHANGE INCLUDED"
+	ewarn "This release adds enforcement of the Taproot protocol change to the Bitcoin"
+	ewarn "rules, beginning in November. Protocol changes require user consent to be"
+	ewarn "effective, and if enforced inconsistently within the community may compromise"
+	ewarn "your security or others! If you do not know what you are doing, learn more"
+	ewarn "before November. (You must make a decision either way - simply not upgrading"
+	ewarn "is insecure in all scenarios.)"
```

Comment:
That kinda implies that it is possible to upgrade without using taproot. But if so, why do we have to `die` if taproot is not enabled?

Reply:
If people actually reject Taproot, they can indeed reject it. But doing so would involve writing code, getting others to adopt it, etc - it's not something we can just support as-is today until someone puts the work into it.

Reply:
> I am currently leaning towards a short warning message on every update, but in case of taproot, a news item seems also sensible. Other means could be discussed as well, like a guard file, that if not existent prevents bitcoind from starting.

I'm on board with a log message and/or news item, but I won't merge with the blocking use flag approach.

I'd really like to get this Bitcoin version bump in soon...

Reply:
In future, I'd really prefer it if a solution was figured out upstream and we didn't need to have these debates. But yeah, agreed. See https://github.com/gentoo/gentoo/pull/21490#discussion_r664209806.